internal/provider/kubernetes/rbac.go

```diff
@@ -3,4 +3,7 @@ package kubernetes
 // +kubebuilder:rbac:groups="gateway.networking.k8s.io",resources=gatewayclasses;gateways;httproutes;referencepolicies;referencegrants,verbs=get;list;watch
 // +kubebuilder:rbac:groups="gateway.networking.k8s.io",resources=gatewayclasses/status;gateways/status;httproutes/status,verbs=update
 
-// +kubebuilder:rbac:groups="",resources=secrets;services;namespaces,verbs=get;list;watch
+// RBAC for Infra Manager to manage Envoy.
```

Comment:
should these live inside `internal/infrastructure/kubernetes` ?

Reply:
Personally I like having them all in one place, but I can live with whatever.

Reply:
https://github.com/envoyproxy/gateway/pull/292 takes the approach of splitting RBAC between Kube provider and Kube infra manager.